File: command-center-tauri/src/utils/__tests__/status.test.ts
import { describe, expect, it } from "vitest";
import { isTerminalWorkflowRunStatus, statusBadgeClass, statusClassForNode } from "../status";

describe("status utils", () => {
  it("maps terminal failures", () => {
    expect(statusBadgeClass("failed")).toBe("status-failed");
    expect(statusClassForNode("timed_out")).toBe("node-danger");
  });

  it("maps active statuses", () => {
    expect(statusBadgeClass("running")).toBe("status-running");
    expect(statusBadgeClass("queued")).toBe("status-waiting");
    expect(statusBadgeClass("debug_paused")).toBe("status-waiting");
    expect(statusClassForNode("debug_paused")).toBe("node-warning");
  });

  it("identifies terminal workflow run statuses", () => {
    expect(isTerminalWorkflowRunStatus("succeeded")).toBe(true);
    expect(isTerminalWorkflowRunStatus("failed")).toBe(true);
    expect(isTerminalWorkflowRunStatus("timed_out")).toBe(true);
    expect(isTerminalWorkflowRunStatus("canceled")).toBe(true);
    expect(isTerminalWorkflowRunStatus("blocked")).toBe(false);
    expect(isTerminalWorkflowRunStatus("running")).toBe(false);
  });
});

File: runinator-command-center-flutter/test/core/utils/status_test.dart
import 'package:runinator_command_center_flutter/core/utils/status.dart';
import 'package:test/test.dart';

void main() {
  group('status utils', () {
    test('maps terminal failures', () {
      expect(statusBadgeClass('failed'), 'status-failed');
      expect(statusClassForNode('timed_out'), 'node-danger');
    });

    test('maps active statuses', () {
      expect(statusBadgeClass('running'), 'status-running');
      expect(statusBadgeClass('queued'), 'status-waiting');
      expect(statusBadgeClass('debug_paused'), 'status-waiting');
      expect(statusClassForNode('waiting'), 'node-waiting');
      expect(statusClassForNode('approval_required'), 'node-waiting');
      expect(statusClassForNode('debug_paused'), 'node-warning');
    });

    test('identifies terminal workflow run statuses', () {
      expect(isTerminalWorkflowRunStatus('succeeded'), isTrue);
      expect(isTerminalWorkflowRunStatus('failed'), isTrue);
      expect(isTerminalWorkflowRunStatus('timed_out'), isTrue);
      expect(isTerminalWorkflowRunStatus('canceled'), isTrue);
      expect(isTerminalWorkflowRunStatus('blocked'), isFalse);
      expect(isTerminalWorkflowRunStatus('running'), isFalse);
    });
  });
}
